File: server/Models/Poem.cs
namespace pbj.Models;

public class Poem : RepoItem<int>
{
    public string Title { get; set; }
    public string Body { get; set; }
    public int Views { get; set; }
    public string Tags { get; set; }
    public int Likes { get; set; }       // cached like count
    public int Saves { get; set; }       // cached save count

    public bool IsArchived { get; set; } // from DB (tinyint(1), default false)
    public bool IsLiked { get; set; }    // computed per user (not stored in poem table)

    public string AuthorId { get; set; } // FK → accounts.id


    /// <summary>
    /// The Profile object for the author of this poem.
    /// </summary>
    public Profile Creator { get; set; }

    /// <summary>
    /// Genres this poem is tagged with (via poemGenre join table).
    /// </summary>
    public List<string> Genres { get; set; } = new();
}

File: client/src/models/Poem.js
import { DBItem } from './DBItem.js'
import { Profile } from './Profile.js'
import { LikedPoem } from './LikedPoem.js'

export class Poem extends DBItem {
  constructor(data) {
    super(data)

    this.id = data.id
    this.title = data.title
    this.body = data.body
    this.views = data.views
    this.tags = data.tags
    this.genre = data.genre

    this.saves = data.saves
    this.isArchived = data.isArchived

    // ✅ keep the DB like-count separate from liked objects
    this.likeCount = data.likes ?? 0  

    this.isLiked = data.isLiked ?? false

    this.authorId = data.authorId

    /** @type {Profile} */
    this.creator = data.creator

    /** @type {LikedPoem[]} */
    this.likedBy = data.likedBy ?? []   // array of users who liked it
  }
}
